packages/token/src/common/erc2981/erc2981.cairo

```diff
@@ -0,0 +1,253 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.15.1 (token/common/erc2981/erc2981.cairo)
+
+/// # ERC2981 Component
+///
+/// Implementation of the NFT Royalty Standard, a standardized way to retrieve royalty payment
+/// information.
+///
+/// Royalty information can be specified globally for all token ids via `set_default_royalty`,
+/// and/or individually for specific token ids via `set_token_royalty`. The latter takes precedence
+/// over the first.
+///
+/// Royalty is specified as a fraction of sale price. The denominator is set by the contract by
+/// using the Immutable Component Config pattern.
+/// See https://community.starknet.io/t/immutable-component-config/114434.
+///
+/// IMPORTANT: ERC-2981 only specifies a way to signal royalty information and does not enforce its
+/// payment. See https://eips.ethereum.org/EIPS/eip-2981#optional-royalty-payments[Rationale] in the
+/// ERC. Marketplaces are expected to voluntarily pay royalties together with sales.
+#[starknet::component]
+pub mod ERC2981Component {
+    use core::num::traits::Zero;
+    use openzeppelin_introspection::src5::SRC5Component::InternalTrait as SRC5InternalTrait;
+    use openzeppelin_introspection::src5::SRC5Component::SRC5Impl;
+    use openzeppelin_introspection::src5::SRC5Component;
+    use openzeppelin_token::common::erc2981::interface::{IERC2981, IERC2981_ID};
+    use starknet::ContractAddress;
+    use starknet::storage::Map;
+
+    // This default denominator is only used when the DefaultConfig
+    // is in scope in the implementing contract.
+    pub const DEFAULT_FEE_DENOMINATOR: u128 = 10_000;
+
+    #[derive(Serde, Drop, starknet::Store)]
+    struct RoyaltyInfo {
+        pub receiver: ContractAddress,
+        pub royalty_fraction: u128,
+    }
+
+    #[storage]
+    struct Storage {
+        default_royalty_info: RoyaltyInfo,
+        token_royalty_info: Map<u256, RoyaltyInfo>,
+    }
+
+    mod Errors {
+        pub const INVALID_ROYALTY: felt252 = 'ERC2981: invalid royalty';
+        pub const INVALID_ROYALTY_RECEIVER: felt252 = 'ERC2981: invalid receiver';
+        pub const INVALID_FEE_DENOMINATOR: felt252 = 'Invalid fee denominator';
+    }
+
+    /// Constants expected to be defined at the contract level used to configure the component
+    /// behaviour.
+    ///
+    /// - `FEE_DENOMINATOR`: The denominator with which to interpret the fee set in
+    ///   `set_token_royalty` and `set_default_royalty` as a fraction of the sale price.
+    pub trait ImmutableConfig {
+        const FEE_DENOMINATOR: u128;
+    }
```

Comment:
I think we should also include a block in the erc2981 component doc section for `ImmutableConfig` (like we do with hooks). IMO this is more pressing now because we're disallowing `FEE_DENOMINATOR` as `0` upon construction and this isn't documented anywhere...it should probably be documented both here and in `initializer`'s description as well so it's clear